src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -52,11 +66,13 @@ public KafkaMonitor(Map<String, Map> testProps) throws Exception {
         _services.put(name, service);
       }
     }
-
-    _executor = Executors.newSingleThreadScheduledExecutor();
   }
 
-  public void start() {
+  public synchronized void start() {
+    if (_isShutdown) {
```

Comment:
We don't know how other people will integrate this with their own container.  It makes sense to me to just make the default behavior safe.